File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-py/models/order_detail.py
from __future__ import annotations

from pydantic import BaseModel


class OrderDetail(BaseModel):
    orderDetailId: int
    orderId: int
    productId: int
    quantity: int
    unitPrice: float
    notes: str | None = None


class CreateOrderDetailRequest(BaseModel):
    orderId: int
    productId: int
    quantity: int
    unitPrice: float
    notes: str | None = None


class UpdateOrderDetailRequest(BaseModel):
    orderId: int | None = None
    productId: int | None = None
    quantity: int | None = None
    unitPrice: float | None = None
    notes: str | None = None

File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-cs/Models/OrderDetail.cs
namespace OctoSupply.Api.Models;

public sealed record OrderDetail(
    int OrderDetailId,
    int OrderId,
    int ProductId,
    int Quantity,
    double UnitPrice,
    string? Notes);

public sealed class CreateOrderDetailRequest
{
    public int OrderId { get; set; }
    public int ProductId { get; set; }
    public int Quantity { get; set; }
    public double UnitPrice { get; set; }
    public string? Notes { get; set; }
}

public sealed class UpdateOrderDetailRequest
{
    public int? OrderId { get; set; }
    public int? ProductId { get; set; }
    public int? Quantity { get; set; }
    public double? UnitPrice { get; set; }
    public string? Notes { get; set; }
}
